Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp

```diff
@@ -369,6 +373,11 @@ void LayerTreeHost::frameComplete()
     m_compositor->frameComplete();
 }
 
+const WebCore::Settings& LayerTreeHost::settings()
+{
+    return m_webPage.corePage()->settings();
+}
```

Comment:
Are Settings thread safe?

Reply:
Apparently not. I guess passing the two flags so they are copied, but then we lose the ability to flip the settings at runtime. Any suggestion on how to go about this?

Reply:
I would use an enum for the two flags, that is passed to the threaded compositor constructor. Then you can define a function ot be called when the settings change with `webcoreOnChange: functionName` in UnifiedWebPreferences.yaml, it could be the same function for both settings if we use the same flags enum. So, when it changes the threaded compositor is notified about the new values.

Reply:
I didn't bother with this in the end. I think it's okay if only the new web views that are created after changing the settings get the new configuration.